metronome/checkpointing/models/src/io/iohk/metronome/checkpointing/models/Mempool.scala

```diff
@@ -0,0 +1,15 @@
+package io.iohk.metronome.checkpointing.models
+
+import io.iohk.metronome.checkpointing.models.Transaction.ProposerBlock
+
+case class Mempool(
+    proposerBlocks: IndexedSeq[ProposerBlock],
+    hasNewCheckpointCandidate: Boolean
+) {
+  def isEmpty: Boolean =
+    proposerBlocks.isEmpty && !hasNewCheckpointCandidate
+}
+
+object Mempool {
+  val empty: Mempool = Mempool(Vector.empty, false)
```

Comment:
It might be safer to start with `true`, just in case for example the PoW network would be waiting for the checkpoint to arrive (like Advocate would), unable to generate new events and not sending new notifications.